### ISSUE TITLE
[Feature Request] super project pointer in submodule log

### ISSUE THREAD

**Issue by twji** [ForkIssues/TrackerWin#575]
Say a parent repo points to its submodule's commit `abcdefg`.
In the submodule's log, I want to see commit `abcdefg` marked by a "virtual tag".
In other words, I want to see the purple label in the following image:
![](https://gitlab.com/tortoisegit/tortoisegit/uploads/d6d9727076e38a643396f92835ff2757/submodule-pointer.png)

So a port of the same feature from TortoiseGit
[https://gitlab.com/tortoisegit/tortoisegit/issues/2826](https://gitlab.com/tortoisegit/tortoisegit/issues/2826)

**Comment by DanPristupov** [comment]
Isn't the current submodule pointer highlighted with bold (because parent always points to HEAD)?

**Comment by twji** [comment]
Yes the currently checked out commit in the submodule would be highlighted. This feature request is about something else. Here's the scenario:

Starting with a parent repo and a submodule both without uncommited changes, and the submodule is at commit `abcd1234`. Then in submodule I make a new commit `abcd5678`.

The result is that
- In submodule's log, `abcd5678` would be in bold.
- Parent repo's "Changes" page would show an unstaged change: that submodule was changed from `abcd1234` to `abcd5678`.

In addition to the current (1.42.4.0) behaviour, I would like commit `abcd1234` in submodule's log to be marked with a "virtual tag". Essentially this is making submodule's log show what parent repo's "Changes" page is showing.

The "virtual tag" could look like a normal tag that is named "superproject pointer". "superproject" being the term used in git documentation.
